testscript/testscript_test.go

```diff
@@ -113,6 +113,9 @@ func TestScripts(t *testing.T) {
 				if ts.Value("somekey") != 1234 {
 					ts.Fatalf("test-values did not see expected value")
 				}
+				if ts.Value("t").(T) != ts.t {
```

Comment:
I think it would be worth checking that it actually implements `testing.TB`, as documented.
```suggestion
				if ts.Value("t").(testing.TB) != ts.t {
```

Reply:
Done, in a slightly different way.